src/routes/transactions/helpers/swap-order.helper.ts

```diff
@@ -0,0 +1,132 @@
+  SwapsRepository,
+} from '@/domain/swaps/swaps.repository';
+import { Token } from '@/domain/tokens/entities/token.entity';
+import { Order } from '@/domain/swaps/entities/order.entity';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { SwapsRepositoryModule } from '@/domain/swaps/swaps-repository.module';
+
+@Injectable()
+export class SwapOrderHelper {
+  private readonly swapsExplorerBaseUri: string =
+    this.configurationService.getOrThrow('swaps.explorerBaseUri');
+
+  constructor(
+    private readonly multiSendDecoder: MultiSendDecoder,
+    private readonly setPreSignatureDecoder: SetPreSignatureDecoder,
+    @Inject(ITokenRepository)
+    private readonly tokenRepository: ITokenRepository,
+    @Inject(ISwapsRepository) private readonly swapsRepository: SwapsRepository,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+  ) {}
+
+  /**
+   * Finds the swap order in the transaction data.
+   * The swap order can be in the transaction data directly or in the data of a Multisend transaction.
+   *
+   * @param data - The transaction data
+   * @private
+   * @returns The swap order if found, otherwise null
+   */
+  public findSwapOrder(data: `0x${string}`): `0x${string}` | null {
+    // The swap order can be in the transaction data directly
+    if (this.isSwapOrder({ data })) {
+      return data;
+    }
+    // or in the data of a multisend transaction
+    if (this.multiSendDecoder.helpers.isMultiSend(data)) {
+      const transactions = this.multiSendDecoder.mapMultiSendTransactions(data);
+      // TODO If we can build a sorted hash map of the transactions, we can avoid iterating all of them
+      //  as we know the pattern of a Swap Order.
+      for (const transaction of transactions) {
+        if (this.isSwapOrder(transaction)) {
+          return transaction.data;
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Retrieves detailed information about a specific order and its associated tokens
+   *
+   * @param {Object} args - The arguments required to fetch the order.
+   * @param {string} args.chainId - The network chain ID.
+   * @param {string} args.orderUid - The unique identifier of the order, prefixed with '0x'.
+   * @returns {Promise} A promise that resolves to an object containing the order and token details.
+   *
+   * The returned object includes:
+   * - `order`: An object representing the order.
+   * - `sellToken`: The Token object with a mandatory `decimals` property
+   * - `buyToken`: Similar to `sellToken`, for the token being purchased in the order.
+   *
+   * @throws {Error} Throws an error if the order `kind` is 'unknown'.
+   * @throws {Error} Throws an error if either the sellToken or buyToken object has null decimals.
+   */
+  async getOrder(args: { chainId: string; orderUid: `0x${string}` }): Promise<{
+    order: Omit<Order, 'kind'> & { kind: Exclude<Order['kind'], 'unknown'> };
```

Comment:
```suggestion
    order: Order & { kind: Exclude<Order['kind'], 'unknown'> };
```

Reply:
Interesting, at some point this was not working as I expected but maybe that was when I didn't have `{ ...order, kind: order.kind }`.

The original idea was to remove the `kind` completely from the order and add a new one without `unknown` (thus the two separate steps here) – i.e. I didn't want all the `Order.kind` +  `Order.kind except 'unknown'`.

Reply:
Both approaches are correct but I noticed that the tokens use the "other" method so I suggested this for consistency. Afaiu, it overwrites the former. Prior to my suggestion, it is more explicit. I would suggest we use one or the other.